tests/dag_processing/test_dag_bundles.py

```diff
@@ -160,6 +184,16 @@ def test_correct_repo_urls(self, conn_id, expected_repo_url):
         hook = GitHook(git_conn_id=conn_id)
         assert hook.repo_url == expected_repo_url
 
+    def test_given_both_private_key_and_key_file(self):
+        with pytest.raises(
+            AirflowException, match="Both 'key_file' and 'private_key' cannot be provided at the same time"
+        ):
+            GitHook(git_conn_id=CONN_BOTH_PATH_INLINE)
+
+    def test_only_inline_connection_has_tmp_keyfile(self):
```

Comment:
We should beef up our coverage on this path too. "how" depends how we refactor I think, but reach out if you need ideas :)

Reply:
Looks good now I think.